docs/DETECTION_AND_CODE_GENERATION.md

```diff
@@ -0,0 +1,91 @@
+## Detection
+
+### Inspection of site
+
+Alltheplaces provides rudimentary tooling to inspect a site for common behaviours.
+
+* [Sitemap inspection](SITEMAP.md) - `pipenv run scrapy sitemap http://example.com/` to detect potential individual store URLs.
+* [Structured Data inspection](STRUCTURED_DATA.md) - `pipenv run scrapy sitemap http://example.com/` to detect potential individual store URLs.
+* Links - `pipenv run scrapy links http://example.com/` to look for links with human language labels, such as "Find our stores".
+
+When checking a large number of URLs, these tools can be chained together to highlight potential candidates for spidering.
+
+### Automatic detection from storefinder page
+
+Alltheplaces has a number of key storefinders which gather together common functionality.
+
+Going one step further, many of these store finders build in the capabilities to automatically detect the
+presence of a common storefinder from either:
+
+- Request patterns made to API endpoints or
+- Responses that contain xpath or JS Objects indicating the presence of a store locator.
```

Comment:
Suggest "Responses in HTML format (DOM elements and JavaScript objects), including any part of a response which is loaded from external JavaScript files or from within iframes)"